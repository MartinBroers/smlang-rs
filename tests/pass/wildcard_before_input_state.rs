extern crate smlang;

use smlang::statemachine;

statemachine! {
    transitions: {
        _ + Event1 = Fault, //~ State and event combination specified multiple times, remove duplicates.
        *State1 + Event1 = State2,
    }
}

fn main() {}
