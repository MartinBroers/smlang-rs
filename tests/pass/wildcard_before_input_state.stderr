warning: unreachable statement
  --> tests/compile-fail/wildcard_before_input_state.rs:5:1
   |
5  | / statemachine! {
6  | |     transitions: {
7  | |         _ + Event1 = Fault, //~ State and event combination specified multiple times, remove duplicates.
8  | |         *State1 + Event1 = State2,
9  | |     }
10 | | }
   | | ^
   | | |
   | |_unreachable statement
   |   any code following this expression is unreachable
   |
   = note: `#[warn(unreachable_code)]` on by default
   = note: this warning originates in the macro `statemachine` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 1 warning emitted
